config/configclientauth/configoauth2.go

```diff
@@ -0,0 +1,90 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configclientauth
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+	"google.golang.org/grpc/credentials"
+	grpcOAuth "google.golang.org/grpc/credentials/oauth"
+)
+
+var (
+	errNoClientIDProvided     = errors.New("no ClientID provided in OAuth Client Credentials configuration")
```

Comment:
```suggestion
	errNoClientIDProvided     = errors.New("no ClientID provided in the OAuth2 exporter configuration")
```

The end-user is going to see this, and it should be very clear what this is about. When doing a manual test, make sure the exporter name is included as a field in the logs (should be automatically added already).